image-compressor/src/main/java/io/mosip/image/compressor/sdk/service/ImageCompressionService.java

```diff
@@ -87,22 +83,54 @@ public Response<BiometricRecord> getExtractTemplateInfo() {
 				 * segment.getBirInfo().setPayload(segment.getBdb());
 				 */
 
-				BDBInfo bdbInfo = segment.getBdbInfo();
-				if (bdbInfo != null) {
-					// Update the level to RAW
-					bdbInfo.setLevel(getProcessedLevelType());
+				if (segment.getBdbInfo() != null) {
```

Comment:
Can we not do the and conditions?

Reply:
Update the condition too and,
added error code if bdbinfo or format is null